### PR TITLE
Feature/18 advance filter

### PR DESCRIPTION
**Pull request by navins94**
- #18

### CODE REVIEW

File: backend/alembic/versions/2023-01-10-07-24-50_e47196be4c5c_add_materialize_view_for_advance_filter.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+revision = "e47196be4c5c"
+down_revision = "d37375e79c27"
+branch_labels = None
+depends_on = None
+
+advance_filter_with_multiple_option_value = PGMaterializedView(
+    schema="public",
+    signature="advance_filter",
+    definition="""
+    SELECT tmp.data,
```
Comment (wayangalihpratama) [inline backend/alembic/versions/2023-01-10-07-24-50_e47196be4c5c_add_materialize_view_for_advance_filter.py:22]: The view does not work as we need when filtering with questions from the monitoring form.
I think, here we need to add `identifier` value from `data` table. So we need to alter this view to include `identifier` column.

File: backend/db/crud_data.py [hunk 2]
```diff
@@ -91,8 +92,22 @@ def get_data(
...
+    options: List[str] = None,
+) -> DataDict:
+    data = session.query(Data).filter(Data.registration == registration)
+    if options:
+        # support multiple select options filter
+        # change query to filter data by or_ condition
+        or_query = or_(
+            ViewAdvanceFilter.options.contains([opt]) for opt in options
+        )
+        data_id = session.query(ViewAdvanceFilter.data).filter(or_query).all()
```
Comment (wayangalihpratama) [inline backend/db/crud_data.py:107]: Here also we need to change the query after fixing `advance_filter` materialized view.
The query should get the `identifier` value:
`data_id = session.query(ViewAdvanceFilter.identifier).filter(or_query).all()`

File: backend/db/crud_data.py [hunk 3]
```diff
@@ -91,8 +92,22 @@ def get_data(
...
+        data = data.filter(Data.id.in_([d.data for d in data_id]))
```
Comment (wayangalihpratama) [inline backend/db/crud_data.py:108]: Same here, we need to change the data filter from `id.in` to `identifier.in`:
`data = data.filter(Data.identifier.in_([d.identifier for d in data_id]))`

File: frontend/src/App.js [hunk 4]
```diff
@@ -2,12 +2,14 @@ import React from "react";
...
 import { Row, Col, Layout } from "antd";
 import { Map } from "./components";
+import AdvanceFilter from "./components/filter";
 
 const { Header, Footer, Content } = Layout;
 
 const App = () => {
   return (
     <Layout className="layout">
+      <AdvanceFilter />
```
Comment (wayangalihpratama) [inline frontend/src/App.js:12]: Maybe we can move Advance Filter below the page header.

**Review by wayangalihpratama — CHANGES_REQUESTED** [review]

File: backend/db/crud_data.py (hunk 3 above)
Comment (navins94) [reply backend/db/crud_data.py]: Could you check the latest commits

**Review by wayangalihpratama — APPROVED** [review]

**Pull request merged**